url28dfd00382_byang223_TurnBAPI_tgz-pJ8-units_upgrades_UpgradeTestJ8/src/units/interactions/PassivePoisonGas.java

```diff
@@ -0,0 +1,65 @@
+package units.interactions;
+
+import java.util.ArrayList;
+
+import map.Tile;
+import modes.models.GameModel;
+import units.Unit;
+import attribute.AttributeAttack;
+import attribute.AttributeExperience;
+import attribute.AttributeHealth;
+
+/**
+ * Basic implementation of a passive poison gas attack. At the end of each turn,
+ * it will do damage to all the units within this unit's AttackAttribute
+ * attackRange.
+ * 
+ * @author Matthew
+ * 
+ */
+public class PassivePoisonGas extends InteractionPassiveTurn {
+
+    @Override
+    public void performInteraction(GameModel g, Unit myUnit) {
+        if (!myUnit.hasAttribute("Attack"))
+            return;
+        AttributeAttack att = (AttributeAttack) myUnit.getAttribute("Attack");
+        ArrayList<Tile> tileList = g.getMap().getTilesList();
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem:**
 This line of code lacks validation when processing input data through the following parameter: &#39;g&#39; (index: 0 | type: GameModel). The parameter is exposed to external callers, because its enclosing class and method are publicly accessible. This means that upstream validation, if it exists, can be bypassed. Other validated parameters: &#39;myUnit&#39;. The same parameter type is validated here for example: url28dfd00382_byang223_TurnBAPI_tgz-pJ8-units_upgrades_UpgradeTestJ8/src/ai/state/AIstate.java:30. Malicious, malformed, or unbounded inputs can cause unexpected runtime behavior or crashes, and can slow performance.

**Fix:**
 Add checks to ensure the validity of the parameter&#39;s value, such as testing it for nullness (for example, using the @NonNull annotation as [described in the Lombok library](https://projectlombok.org/features/NonNull)), emptiness, or equality. Or to prevent direct calls to it, [reduce the method&#39;s visibility](https://docs.oracle.com/javase/tutorial/java/javaOO/accesscontrol.html).

 Learn more about potential threats and guidance from the [Common Weakness Enumeration website](https://cwe.mitre.org/data/definitions/693.html) and the [OWASP Cheat Sheet series](https://cheatsheetseries.owasp.org/cheatsheets/Input_Validation_Cheat_Sheet.html).